src/controller/CHIPDeviceController.cpp

```diff
@@ -677,8 +675,8 @@ CHIP_ERROR DeviceCommissioner::EstablishPASEConnection(NodeId remoteDeviceId, Re
     session = mSystemState->SessionMgr()->CreateUnauthenticatedSession(params.GetPeerAddress(), device->GetMRPConfig());
     VerifyOrExit(session.HasValue(), err = CHIP_ERROR_NO_MEMORY);
 
-    err = mSystemState->SessionIDAlloc()->Allocate(keyID);
-    SuccessOrExit(err);
+    secureSessionHolder = mSystemState->SessionMgr()->AllocateSession();
+    VerifyOrExit(secureSessionHolder, err = CHIP_ERROR_NO_MEMORY);
```

Comment:
Seems like this existing split where the device controller allocates the session, then passes that into `Pair` is a bit un-necessary. We should just do the allocation closer to the logic that actually owns/makes use of that session.

Reply:
I absolutely agree and have pushed an update that does this.